### PR TITLE
Fix examples to run from checkout and with newer codes

### PR DESCRIPTION
**Pull request by kiwiroy**
The examples lead to errors when running from a checkout.

```
Unknown command "webpack", maybe you need to install it?
```

Various `webpack` and `rollup` config/plugin related errors are also fixed, including #4.

### CODE REVIEW

File: example/rollup/lite.pl [hunk 1]
```diff
@@ -1,20 +1,37 @@
...
+}
 
 __DATA__
 @@ index.html.ep
 <!DOCTYPE html>
 <html>
   <head>
     <title>Mojolicious ♥ Webpack</title>
-    %= asset 'example.css'
+    % # asset 'example.css'
```
Comment (jhthorsen) [inline example/rollup/lite.pl:29]: Is this supposed to be part of this PR?

File: example/rollup/lite.pl [hunk 2]
```diff
@@ -1,20 +1,37 @@
...
 
-plugin Webpack =>
-  {process => ['svelte'], dependencies => {core => 'rollup', svelte => [qw(rollup-plugin-svelte svelte)]}};
+$ENV{MOJO_WEBPACK_CONFIG}  = "@{[curfile->sibling('rollup.config.js')]}";
+$ENV{MOJO_WEBPACK_DEBUG}   = 1;
+$ENV{MOJO_WEBPACK_VERBOSE} = 0;
+$ENV{MOJO_WEBPACK_LAZY}    = 0;
+
+my $path = "@{[curfile->dirname->dirname->sibling('lib')]}";
+lib->import($path);
```
Comment (jhthorsen) [inline example/rollup/lite.pl]: Not sure if I like the `$path` variable name. Could you change it to `$lib_path` or something instead?

File: example/rollup/lite.pl [hunk 3]
```diff
@@ -1,20 +1,37 @@
...
-BEGIN { unshift @INC, "$FindBin::Bin/../../lib" }
+use Mojo::File 'curfile';
+require lib;
 
-plugin Webpack =>
-  {process => ['svelte'], dependencies => {core => 'rollup', svelte => [qw(rollup-plugin-svelte svelte)]}};
+$ENV{MOJO_WEBPACK_CONFIG}  = "@{[curfile->sibling('rollup.config.js')]}";
+$ENV{MOJO_WEBPACK_DEBUG}   = 1;
+$ENV{MOJO_WEBPACK_VERBOSE} = 0;
+$ENV{MOJO_WEBPACK_LAZY}    = 0;
```
Comment (jhthorsen) [inline example/rollup/lite.pl]: Can you explain why all of these environment variables are required now?

File: example/rollup/lite.pl [hunk 4]
```diff
@@ -1,20 +1,37 @@
...
+  dependencies => {
+    core   => 'rollup',
+    svelte => [qw(rollup-plugin-svelte svelte)]
+  }
+};
 get '/' => 'index';
-app->start;
+
+{
+  local $ENV{PERL5OPT} = "-I$path";
```
Comment (jhthorsen) [inline example/rollup/lite.pl]: Can you add a comment for when this environment variable is needed?

File: example/rollup/rollup.config.js [hunk 5]
```diff
@@ -1,11 +1,11 @@
 import commonjs from 'rollup-plugin-commonjs';
-import resolve from 'rollup-plugin-node-resolve';
+import resolve from '@rollup/plugin-node-resolve';
 import {terser} from 'rollup-plugin-terser';
 
 // The html plugin generates a file that M::P::Webpack reads to generate the
 // correct output when calling <%= asset 'example.js' %> and
 // <%= asset 'example.css' %>
-import html from 'rollup-plugin-bundle-html';
+const html = require('@rollup/plugin-html');
```
Comment (jhthorsen) [inline example/rollup/rollup.config.js]: Why is this changed from `import` to `const`?

**Review by jhthorsen — CHANGES_REQUESTED** [review]
Looks like this PR includes two changes: One is to fix all the dependencies and one for @INC fixes..? Can you split and/or clarify the different changes?

File: example/rollup/lite.pl (hunk 1 above)
Comment (kiwiroy) [reply example/rollup/lite.pl:29]: Yeah, but happy to find a solution that either evaluates it, or removes the line completely.
Currently raises an exception:

        Unknown asset name "example.css". at mojolicious-plugin-webpack/lib/Mojolicious/Plugin/Webpack.pm line 80.

File: example/rollup/lite.pl (hunk 3 above)
Comment (kiwiroy) [reply example/rollup/lite.pl]: `MOJO_WEBPACK_CONFIG` is required otherwise `Mojolicious::Plugin::Webpack` decides it is a `webpack` project rather than `rollup`. Others are removed as they were added in haste.

File: example/rollup/lite.pl (hunk 4 above)
Comment (kiwiroy) [reply example/rollup/lite.pl]: I've added POD to detail how I ran these examples as a new user - generally the `carton` route. I don't generally install, and the code assumes `M::P::Webpack` isn't installed hence the need for the `use lib ...;`. `@INC` changes aren't propagated unless I'm missing something. Without it `mojo` isn't able to find `Mojolicious::Command::Author::webpack`.

https://github.com/jhthorsen/mojolicious-plugin-webpack/blob/5a397cd07b7fe21d262af4fd1d3d1ca9a6ed6926/lib/Mojolicious/Command/Author/webpack.pm#L40-L45

File: example/rollup/rollup.config.js (hunk 5 above)
Comment (kiwiroy) [reply example/rollup/rollup.config.js]: An in haste copy-paste.

**Comment by kiwiroy** [comment]
> Looks like this PR includes two changes: One is to fix all the dependencies and one for @inc fixes..? Can you split and/or clarify the different changes?

I prepared this a little in haste and although it's a bit tidier now, there are still the two components you mention. With a little git trickery I can probably split them and check with `PERL5OPT` set appropriately. Let me know.
